share/completions/jj.fish

```diff
@@ -1 +1,10 @@
-jj util completion fish | source
+# The reason for `__this-command-does-not-exist` is that, if dynamic completion
+# is not implemented, we'd like to get an error reliably. However, the
+# behavior of `jj` without arguments depends on the value of a config, see
+# https://jj-vcs.github.io/jj/latest/config/#default-command
+if set -l completion (COMPLETE=fish jj __this-command-does-not-exist 2>/dev/null)
```

Comment:
Using `jj 0.23.0-0ca6f00421f6f893e974e0f5349e126abd41a812`

Pressing tab on

`$ jj --`


prints a spurious

`Hint: Use `jj -h` for a list of available commands.`

Similar on `jj --repository /`.

Reply:
AFAICT the only remaining regression is https://github.com/clap-rs/clap/pull/5949

I'm planning to merge this PR after that clap fix is in jj. Though it may be worth merging before that, not sure